src/UXClient/Components/LineChart/LineChart.ts

```diff
@@ -1286,9 +1287,71 @@ class LineChart extends TemporalXAxisComponent {
         this.chartOptions.swimLaneOptions = this.originalSwimLaneOptions;
     }
 
+    private createSwimlaneLabels(offsetsAndHeights, visibleCDOs){
+        // Create swimlane label obj
+        let swimLaneLabels = {};
+
+        // Creates object entry for each swimlane with offset, height, and label
+        visibleCDOs.forEach((aggGroup, i) => {
+            let swimLane = aggGroup.swimLane;
+            if(!(aggGroup.swimLane in swimLaneLabels)){ // Only add swimlanes once to swimLaneLabels map
+                swimLaneLabels[aggGroup.swimLane] = {
+                    offset: offsetsAndHeights[i][0],
+                    height: offsetsAndHeights[i][1],
+                    label: this.chartOptions?.swimLaneOptions?.[swimLane]?.label,
+                    onClick: () => this.chartOptions?.swimLaneOptions?.[swimLane]?.onClick?.(swimLane)
+                }
+            } else if(aggGroup.dataType !== DataTypes.Numeric){ // if lane contains non-numeric data and is being added to another lane
+                swimLaneLabels[aggGroup.swimLane].height += offsetsAndHeights[i][1]; // add heights (non-numerics don't share Y axis)
```

Comment:
Should we have the same kind of aggGroup.swimLane in swimLaneLabels check for swimLaneLabels[aggGroup.swimLane] here? Like if exist set the height of it...

Reply:
I think logically, that else if block will only be entered when the swimLaneLabels object contains the swimlane being checked, but it's not at all obvious.  I'll add an additional check just to be safe!